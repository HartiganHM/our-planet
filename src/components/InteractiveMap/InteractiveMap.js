import React, { useState } from 'react';
import { Link } from 'react-router-dom';

import { copyContent, continentMapData, continentsPaths } from '../../data';

import './InteractiveMap.scss';

const { interactiveMap } = copyContent;

const GRID = 'grid';
const OCEANS = 'oceans';
const CONTINENT = 'continent';

const InteractiveMap = () => {
  const [continent, updateContinent] = useState('');

  const handleMouseHover = continent => {
    updateContinent(continent);
  };

  const createPath = ({ className, displayName, link, name }) =>
    className === GRID ? (
      <g id={className}>
        <g transform={continentsPaths[className].transform}>
          <path className={className} d={continentsPaths[className].path} />
        </g>
      </g>
    ) : (
      <g id={className}>
        <g transform={continentsPaths[className].transform}>
          <Link to={link}>
            <path
              onMouseEnter={() => handleMouseHover(displayName || name)}
              onMouseLeave={() => handleMouseHover('')}
              className={className === OCEANS ? className : CONTINENT}
              d={continentsPaths[className].path}
            />
          </Link>
        </g>
      </g>
    );

  const hoverBox = (
    <div className="hover-box">
      <div className="hover-title">{interactiveMap.hoverTitle}</div>
      <div className="hover-body">
        <div>{interactiveMap.hoverBody}</div>
        <div className="hover-continent">
          {continent || interactiveMap.noHoverPlaceholder}
        </div>
      </div>
    </div>
  );

  return (
    <div className="InteractiveMap">
      {hoverBox}
      <svg version="1.1" x="0px" y="0px" viewBox="0 0 2371 1207">
        {Object.keys(continentMapData).map(continent =>
          createPath(continentMapData[continent])
        )}
      </svg>
    </div>
  );
};

export default InteractiveMap;
