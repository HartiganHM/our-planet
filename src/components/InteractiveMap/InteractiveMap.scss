@import '../../styles/variables.scss';
@import '../../styles/mixins.scss';
.InteractiveMap {
  @include width(160);
  @include padding(1);
  margin: auto;
  position: relative;
  .grid {
    fill: $snow;
  }
  .oceans {
    fill: $spanish-gray;
    transition: 0.5s fill ease-in-out;
    &:hover {
      fill: $accent-color-light;
    }
  }
  .continent {
    fill: $isabelline;
    transition: 0.5s fill ease-in-out;
    &:hover {
      fill: $accent-color-light;
    }
  }
  .hover-box {
    @include padding(1);
    margin: auto;
    color: $accent-color-dark;
    .hover-title {
      @include contentDetailTitle();
    }
    .hover-body {
      @include maxWidth(55);
      @include contentDetailBody();
      .hover-continent {
        font-size: 1.8rem;
        text-align: center;
      }
    }
  }
}

@media screen and (max-width: 1634px) {
  .InteractiveMap {
    @include width(128);
  }
}

@media screen and (max-width: 1334px) {
  .InteractiveMap {
    @include width(96);
  }
}

@media screen and (max-width: 1014px) {
  .InteractiveMap {
    @include width(64);
  }
}

@media screen and (max-width: 694px) {
  .InteractiveMap {
    @include width(32);
  }
}
