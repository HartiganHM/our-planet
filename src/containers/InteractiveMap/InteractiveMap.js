import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink, Link } from 'react-router-dom';
import * as actions from '../../actions';
import PropTypes from 'prop-types';
// import './Header.css';

class InteractiveMap extends Component {
  render() {
    const buttons = this.props.continents.map(continent => {
      const continentLink = '/continents/' + continent.name;

      return <Link to={continentLink}>{continent.name}</Link>;
    });

    return <div>{buttons}</div>;
  }
}

export const mapStateToProps = store => {
  return {
    continents: store.continents
  };
};

export default connect(mapStateToProps, null)(InteractiveMap);
