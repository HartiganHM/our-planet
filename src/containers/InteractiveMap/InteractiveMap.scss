.InteractiveMap {
  display: flex;
  justify-content: space-around;
  .continent-button {
    font-size: 1.4rem;
  }
}