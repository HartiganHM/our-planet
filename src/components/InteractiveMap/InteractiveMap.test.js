import React from 'react';
import { shallow } from 'enzyme';
import InteractiveMap from './InteractiveMap';
import continentsPaths from '../../data/continentsPaths';

describe('InteractiveMap tests', () => {
  let renderedMap;

  beforeEach(() => {
    renderedMap = shallow(<InteractiveMap />);
  });

  it('Should match the snapshot', () => {
    expect(renderedMap).toMatchSnapshot();
  });

  it('Should desiplay a link for each continent', () => {
    const expectedElement = 'Link';
    const expectedLength = Object.keys(continentsPaths).length - 1;

    expect(renderedMap.find(expectedElement).length).toEqual(expectedLength);
  });

  it('Should setState with handleMouseHover', () => {
    const expectedState = {
      isHovering: true,
      continent: 'Asia'
    };

    renderedMap.instance().handleMouseHover('Asia');
    expect(renderedMap.state()).toEqual(expectedState);
  });

  it('Should setState with setContinent', () => {
    const expectedContinent = 'Europe';

    renderedMap.instance().setContinent('Europe');
    expect(renderedMap.state('continent')).toEqual(expectedContinent);
  });

  it('Should receive an object from toggleHoverState with an opposite boolean', () => {
    const expectedReturn = { isHovering: true };
    const mockReturnObject = renderedMap
      .instance()
      .toggleHoverState(renderedMap.state());

    expect(mockReturnObject).toEqual(expectedReturn);
  });

  it('Should setState with the return object from toggleHoverState', () => {
    const expectedState = {
      isHovering: true,
      continent: ''
    };

    expect(renderedMap.state('isHovering')).toEqual(false);
    renderedMap.instance().handleMouseHover('');

    expect(renderedMap.state()).toEqual(expectedState);
  });

  it('Should return svg path elements with createPath', () => {
    const mockPath = renderedMap
      .instance()
      .createPath('north-america', 'North America');
    const expectedTransform = continentsPaths['north-america'].transform;
    const expectedPath = continentsPaths['north-america'].path;

    expect(mockPath.props.children.props.transform).toEqual(expectedTransform);
    expect(
      mockPath.props.children.props.children.props.children.props.d
    ).toEqual(expectedPath);
  });

  it('Should return different render when createPath is passed grid', () => {
    const mockPath = renderedMap.instance().createPath('grid');
    const renderedGrid = renderedMap.find('.grid').prop('d');
    const expectedPath = mockPath.props.children.props.children.props.d;

    expect(renderedGrid).toEqual(expectedPath);
  });
});
