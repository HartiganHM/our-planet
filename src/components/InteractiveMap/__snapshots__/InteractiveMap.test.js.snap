// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InteractiveMap tests Should match the snapshot 1`] = `ShallowWrapper {}`;
